Added audio, sensitivity and inverted axis methods. Ditched brightness, for now

bottolo opened the pull request:
### It just works
Unironically, it does. Here are the changes I've done to implement these methods:
- **Added working audio settings**. They'll work throughout the levels and are resilient to the saving mechanic.
- **Added working sensitivity and axis inversion methods**! It's just a multiplication, but it does the trick.
- As a result, I've updated `MouseLook.cs` to implement a sensitivity and axis inversion method.
- I've also done some more polishing in the comments, particularly in `MouseLook.cs`. 
- MenuController and LoadPrefs have been updated to accomodate these changes, and its summary has been revisited.
- Oh, uhh, the menu has a new save icon. It's really cool. I would dare to say it's the best piece of art I've ever seen around since the dawn of men.

### It just doesn't work
- Brightness. The most fearsome enemy a man can fight. I have nightmares about it. From now on I'll live in fear. Please help.
- Jokes aside, brightness has been temporarily ditched. There's an issue regarding this discussion, and if anyone wants to give his feedback, well... it would be appreciated. 

Now, it would be a shame if I had a plethora of conflicts while merging...
.................................
.................................
.................................
................................................. **_right_**?

bottolo committed:
Volume control added

It now works and mantains settings throughout each level

bottolo committed:
Ditched brightness

After hours spent on post processing, I can't get brightness to change. I'll ditch it for now and see if I can make sensitivity and invert axis work.

bottolo committed:
Mouse sensitivity settings now working

I've added a simple snippet in the MouseLook script. If there's a key regarding mouse sensitivity, then apply this by multiplying the sensitivity with the preferred amount.

bottolo committed:
Updated MouseLook once again to invert axis

Now settings are properly linked to their respective functions. You can invert your view axis!

bottolo committed:
Rearranged code before PR

Added summary to MouseLook.cs, where I've described what it does a little more in depth.

bottolo committed:
Deleted a scriptable object folder

bottolo commented:
Oh, look at that. A scene conflict.

bottolo committed:
Deleted scene 3

Trying to resolve conflicts.

bottolo committed:
Revert "Deleted scene 3"

This reverts commit 477ac6a71ea46b3d6057b26dbc12d8fe541a0972.

bottolo committed:
Merge remote-tracking branch 'origin/develop' into bottolo/settings-methods

# Conflicts:
#	Assets/Code/Scripts/PlayerManagement/MouseLook.cs
#	Assets/Level/Scenes/Scene_3_new_respawn_structure.unity

spig95 commented on Assets/Code/Scripts/PlayerManagement/MouseLook.cs:
```diff
@@ -1,20 +1,28 @@
+using Unity.VisualScripting;
 using UnityEngine;
 
 public class MouseLook : MonoBehaviour
 {
+    /// <summary>
+    /// Everything here is linked to how cursor input is processed and used.
+    /// Rotation, orientation, sensitivity handling and eventual added preferences such as inverted vertical axis.
+    /// </summary>
+
```
Picky me: the summary of the class should be above the class definition ;)

bottolo replied:
> - I think that the "obsolete" files in Arts/Background can be removed, right?

I'll check if it can be removed. There should be my placeholder there for the menu, I'll update you once I've checked.

> - If I click on the settings, I can only open the gameplay settings (and not the audio ones, for instance). Is this a bug, or do you plan of implementing this in the future?

This is a bug which Alessio encountered last week, and since you're having the same issue I'll try and see what causes it. Thanks for the feedback.

> - I have tried to thick the invert option, and the mouse behaves the same way (if I move up, the camera moves up, if I move down, the camera moves down). Maybe I did not get the role of that option, but I expected different behaviors

It's entirely my fault. I merged and did not check if the inverted Y continued to work. I'll  debug as soon as possible.

bottolo replied:
> Picky me: the summary of the class should be above the class definition ;)

Yeah, sorry. I'll edit it out in the next commits.

bottolo replied:
I've checked if the Y inversion works, and it does on my end. Did you confirm your settings before playing? 

Also I think the Art folder is still needed, since everything about logos and backgrounds will be stored there. What I can do to tidy up these folders is to move the UI folder into the Art one.

Let me know if you still experience the missing Y inversion. 

Last, but not least, indeed the text is all messed up and button area is not working as intended. Still trying to figure out what it is.

bottolo replied:
Okay, the menu should now be fixed. Inverted Y still works, tried multiple times. I'm going to resolve this conversation, if you still experience this issue let me know.

spig95 replied:
No I do not experience it, I was probably wrong and it was working since before. Anyway, now also the other settings are working! I'll approve

spig95 commented on Assets/Code/Scripts/PlayerManagement/MouseLook.cs:
```diff
@@ -52,12 +67,26 @@ void Update()
+                // The following rotation order is needed because if the object is the same,
+                // the second assign value overwrites the values of the first one. If the object is different, the order is irrelevant.
+                
+                if (PlayerPrefs.GetInt("masterInvertY") == 1)
```
```suggestion
                if (PlayerPrefs.GetInt("masterInvertY"))
```

bottolo replied:
I didn't know you could do that, I'm committing the suggestion.

spig95 reviewed (commented):
Nice description, loved it! The menu works, the music as well! :) Here are my other comments:

- I think that the "obsolete" files in Arts/Background can be removed, right?
- If I click on the settings, I can only open the gameplay settings (and not the audio ones, for instance). Is this a bug, or do you plan of implementing this in the future?
- I have tried to thick the invert option, and the mouse behaves the same way (if I move up, the camera moves up, if I move down, the camera moves down). Maybe I did not get the role of that option, but I expected different behaviors

stefanofossati commented on Assets/Code/Scripts/PlayerManagement/MouseLook.cs:
```diff
@@ -52,12 +67,26 @@ void Update()
             _xRotation = Mathf.Clamp(_xRotation, -90f, 90f);    // Clamping allows to block the rotation
-        
-            // rotate camera and orientation
-            /*This order is needed because if the object is the same, the second assign value overwrites the values of the first one.
```
Insert this comment in one or both the if statements, I have left this comment because this stupid order caused me some troubles

bottolo replied:
I've left it there because I was way out of reach with the current PRs. Nicola said the inverted Y doesn't work either, so I'll definitely have a look into that.

bottolo replied:
Comment deleted, it should be clearer now.

bottolo committed:
Update Assets/Code/Scripts/PlayerManagement/MouseLook.cs

Co-authored-by: Nicola Ischia <36990714+spig95@users.noreply.github.com>

bottolo committed:
Resolved compile error, checked Y inversion, moved summary

bottolo committed:
Moved UI folder to the Art one, fixed menu

Upon given feedback, I've done some checks with the menu and now everything should work accordingly.

TizioMaurizio commented on Assets/Code/Scripts/PlayerManagement/MouseLook.cs:
```diff
@@ -48,16 +60,27 @@ void Update()
-            _transform.rotation = Quaternion.Euler(_xRotation, _yRotation, 0);
+            
+            // Checks whether there's local information about vertical axis preference and changes it.
+            if (PlayerPrefs.HasKey("masterInvertY"))
```
![image](https://user-images.githubusercontent.com/48728364/200438499-5e87b4b7-45b0-4bf0-bd1f-41b605aadb7a.png)

TizioMaurizio replied:
e se non ce l'ha?

bottolo replied:
Too many questions.
